### PR TITLE
Posts with interview Experience, filter and clear button for filters

### PR DESCRIPTION
**Pull request by Nik-git1**
-Implemented Markers for posts with interview Experience(had to modify parsing logic for this), the markers are clickable as well.
-Added filters for post with interview experience
-added "Clear" functionality for filters.
exp1
exp2

Issue : Had to reparse the raw files , but couldn't complete all of them due to API limit(current PR has around 180 posts only) ,and upon restarting the parsing it does not start parsing rest of the raw files , logically it should compare date between latest parsed and raw file , but it is not working , Could you please solve this issue or provide me with a fix?

### CODE REVIEW

**Comment by kuutsav** [comment]
I'll update the data and fix any other issue. Before I do that, please revert the changes to the data files in this PR.

**Comment by Nik-git1** [comment]
Done, let me know if I could be of more help. Thanks.

**Comment by kuutsav** [comment]
@Nik-git1 I'll review it sometime this week. A little tied up atm.

File: script.js [hunk 1]
```diff
@@ -434,17 +458,21 @@ document.addEventListener('DOMContentLoaded', async function () {
...
     }
 
+
     function filterOffersByCompany(companyName) {
         globalFilterState.searchString = companyName;
         filterOffers(); // Call the unified filter function
     }
 
-    function filterOffersByYoeAndSalary(yoeRange, salaryRange) {
+    function FilterMenu(yoeRange, salaryRange, includeInterviewExp) {
```
Comment (kuutsav) [inline script.js:467]: Let's keep the func names consistent, FilterMenu -> filterMenu

File: script.js [hunk 2]
```diff
@@ -395,37 +414,42 @@ document.addEventListener('DOMContentLoaded', async function () {
...
     });
-
     function filterOffers() {
         currentSort = { column: null, order: 'asc' };
 
-        // Start with all offers
         let tempFilteredOffers = [...offers];
 
-        // Filter by search string if applicable
+        // Apply search string filter if applicable
```
Comment (kuutsav) [inline script.js:422]: Can you please remove these comment changes and keep the overall changes in this PR as minimal as possible?

File: leetcomp/parse.py [hunk 3]
```diff
@@ -83,9 +83,18 @@ def parsed_content_is_valid(parsed_content: list[dict[Any, Any]]) -> bool:
     return True  # Parsed content is valid if no assertions fail
 
 
+def extract_interview_exp(content: str) -> str:
+    match = re.search(
+        r"https://leetcode.com/discuss/interview-experience/\S+", content
```
Comment (kuutsav) [inline leetcomp/parse.py:88]: keep this pattern at the top of the file after the imports, interview_exp_pattern = re.compile("...")

File: index.html [hunk 4]
```diff
@@ -94,9 +94,18 @@
                                 <div class="col-auto"><input type="number" id="salaryMax" name="salary-max" class="form-control form-control-sm custom-input-width" min="1" max="200" value="200"></div>
                             </div>
                         </div>
-                        <button id="filter" class="px-2 mt-3 btn btn-primary btn-sm">Apply</button>
+
+                        <div class="mt-3">
+                            <input type="checkbox" id="interviewExpFilterCheckbox">
+                            <label for="interviewExpFilterCheckbox"><b>Filter by Interview Experience</b></label>
```
Comment (kuutsav) [inline index.html:100]: Let's rename it to the "Includes Interview Exprience"

File: script.js [hunk 5]
```diff
@@ -10,13 +10,20 @@ let totalPages = 0;
 const svgWidth = 16;
 const svgHeight = 16;
 
+const starSVG = `
+    <svg fill="#000000" width="24px" height="24px" viewBox="0 0 1024 1024" xmlns="http://www.w3.org/2000/svg" class="icon">
```
Comment (kuutsav) [inline script.js:14]: 24px looks too big, maybe 12 or 14

**Comment by Nik-git1** [comment]
Done; this was a long PR, and I tried my best to minimize the changes.

**Comment by kuutsav** [comment]
> Done; this was a long PR, and I tried my best to minimize the changes.

Yup, Thanks for the patience.

**Comment by kuutsav** [comment]
@Nik-git1 I've made few design changes to keep the ui a little clean. Removed the svg star and using a simple unicode symbol now. Also, did a bulk data update offline to reflect the interview-exp badge now. It should work as is with auto data refresh now.
Thanks again!

**Pull request merged**